src/test/java/com/academia/library/validator/OrderValidatorTest.java

```diff
@@ -0,0 +1,155 @@
+import com.academia.library.exception.OrderStatusException;
+import com.academia.library.model.Book;
+import com.academia.library.model.Order;
+import com.academia.library.model.OrderStatus;
+import com.academia.library.repository.BookRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+class OrderValidatorTest {
+
+    @Mock
+    BookRepository bookRepository;
+
+    @InjectMocks
+    OrderValidator orderValidator;
+
+    @Test
+    void validateBeforeCreateSuccess() {
+        // given
+        var bookId = 1L;
+        var count = 1L;
+        var orderStatus = OrderStatus.DRAFT.name();
+
+        var book = new Book();
+
+        var orderRequest = OrderRequest.builder()
+                .status(orderStatus)
+                .orderDetails(List.of(OrderDetailsRequest.builder()
+                        .orderDetailRequestDto(OrderDetailsRequest.OrderDetailRequestDto.builder()
+                                .bookId(bookId)
+                                .count(count)
+                                .build())
+                        .build()))
+                .build();
+        // when
+        when(bookRepository.findById(bookId)).thenReturn(Optional.of(book));
+        // then
+        orderValidator.validateBeforeCreate(orderRequest);
+
+    }
+
+    @Test
+    void validateBeforeUpdateEntitySuccess() {
+        // given
+        var bookId = 1L;
+        var count = 1L;
+        var orderStatus = OrderStatus.DRAFT;
+
+        var book = new Book();
+
+        var order = new Order();
+        order.setOrderStatus(orderStatus);
+
+        var orderRequest = OrderRequest.builder()
+                .status(orderStatus.name())
+                .orderDetails(List.of(OrderDetailsRequest.builder()
+                        .orderDetailRequestDto(OrderDetailsRequest.OrderDetailRequestDto.builder()
+                                .bookId(bookId)
+                                .count(count)
+                                .build())
+                        .build()))
+                .build();
+        // when
+        when(bookRepository.findById(bookId)).thenReturn(Optional.of(book));
+        // then
+        orderValidator.validateBeforeUpdate(orderRequest, order);
+    }
+
+    @Test
+    void testBeforeUpdateStatusSuccess() {
+        var orderStatus = OrderStatus.PAID;
+        var executeStatus = OrderStatus.PAID;
+
+        var order = new Order();
+        order.setOrderStatus(orderStatus);
+
+        // when, then
+        orderValidator.validateBeforeUpdate(order, executeStatus);
+    }
+
+    @Test
+    void validateBeforeCreateException() {
```

Comment:
validateBeforeCreateException -> validateBeforeCreateExceptionShouldThrowOrderStatusException

And please try to name test methods correctly. Name should tell what this method do

Reply:
Ok, thank